xbmc/GUIPassword.cpp

```diff
@@ -548,13 +548,12 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
   return false;
 }
 
-bool CGUIPassword::IsMediaPathUnlocked(const std::string& strType)
+bool CGUIPassword::IsMediaPathUnlocked(const std::shared_ptr<CProfileManager>& profileManager,
+                                       const std::string& strType)
 {
   if (!StringUtils::StartsWithNoCase(strMediasourcePath, "root") &&
       !StringUtils::StartsWithNoCase(strMediasourcePath, "library://"))
   {
-    const std::shared_ptr<CProfileManager> profileManager =
-        CServiceBroker::GetSettingsComponent()->GetProfileManager();
     if (!g_passwordManager.bMasterUser &&
         !profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
```

Comment:
https://jenkins.kodi.tv/job/Android-ARM/18427/clang/new/source.7fe16e94-41bf-43a3-8c15-087bc1747873/#560

not sure why this generates a new warning, it's unchanged
i'd change to `profileManager->GetMasterProfile().getLockMode() != LOCK_MODE_EVERYONE)`